cohorts/variant_stats.py

```diff
@@ -0,0 +1,141 @@
+from collections import namedtuple
+
+
+VariantStats = namedtuple('VariantStats', 
+                          ['depth', 'alt_depth', 'variant_allele_frequency'])
+
+
+SomaticVariantStats = namedtuple('SomaticVariantStats', 
+                          ['tumor_stats', 'normal_stats'])
+
+
+def strelka_somatic_variant_stats(variant, variant_metadata):
+    """Parse out the variant calling statistics for a given variant from a Strelka VCF
+
+    Parameters
+    ----------
+    variant : varcode.Variant
+    sample_info : dict
+        Dictionary of sample to variant calling statistics, corresponds to the sample columns
+        in a Strelka VCF
+
+    Returns
+    -------
+    SomaticVariantStats
+    """
+
+    sample_info = variant_metadata['sample_info']
+    # Ensure there are exactly two samples in the VCF, a tumor and normal
+    assert len(sample_info) == 2, "More than two samples found in the somatic VCF"
+
+    tumor_stats = _strelka_variant_stats(variant, sample_info['TUMOR'])
+    normal_stats = _strelka_variant_stats(variant, sample_info['NORMAL'])
+    return SomaticVariantStats(tumor_stats=tumor_stats, normal_stats=normal_stats)
+
+def _strelka_variant_stats(variant, sample_info):
+    """Parse a single sample's variant calling statistics based on Strelka VCF output
+
+    Parameters
+    ----------
+    variant : varcode.Variant
+    sample_info : dict
+        Dictionary of Strelka-specific variant calling fields
+
+    Returns
+    -------
+    VariantStats
+    """
+
+    # Retrieve the Tier 1 counts from Strelka
+    ref_depth = int(sample_info[variant.ref+'U'][0])
+    alt_depth = int(sample_info[variant.alt+'U'][0])
+    depth = alt_depth + ref_depth
+    vaf = float(alt_depth) / depth
+
+    return VariantStats(depth=depth, alt_depth=alt_depth, variant_allele_frequency=vaf)
+
+def mutect_somatic_variant_stats(variant, variant_metadata):
+    """Parse out the variant calling statistics for a given variant from a Mutect VCF
+
+    Parameters
+    ----------
+    variant : varcode.Variant
+    sample_info : dict
+        Dictionary of sample to variant calling statistics, corresponds to the sample columns
+        in a Mutect VCF
+
+    Returns
+    -------
+    SomaticVariantStats
+    """
+
+    sample_info = variant_metadata['sample_info']
+    # Ensure there are exactly two samples in the VCF, a tumor and normal
+    assert len(sample_info) == 2, "More than two samples found in the somatic VCF"
+
+    # Find the sample with the genotype field set to variant in the VCF
+    tumor_sample_infos = [info for info in sample_info.values() if info['GT'] == '0/1']
```

Comment:
Unfortunately, that's not a standard. Mutect places the sample names as column headers. So this could really be anything, 'Arun_T1/Arun_N1`, `blood\tumor` ... whatever the samples were named. In the test files, they just happen to be named `T1` and `N1`.

On the other hand, Strelka only uses two possible names, `NORMAL` or `TUMOR`